p2p/conn/connection.go

```diff
@@ -869,7 +984,10 @@ func (ch *Channel) recvPacketMsg(packet tmp2p.PacketMsg) ([]byte, error) {
 		//   suggests this could be a memory leak, but we might as well keep the memory for the channel until it closes,
 		//	at which point the recving slice stops being used and should be garbage collected
 		ch.recving = ch.recving[:0] // make([]byte, 0, ch.desc.RecvBufferCapacity)
-		return msgBytes, nil
+
+		msgCopy := make([]byte, len(msgBytes))
```

Comment:
[To reviewers] Without creating a new copy, the original output of this function `msgBytes` and the `ch.recving` slice would reference the same memory address. Consequently, if `msgBytes` is passed to another goroutine, it can lead to a race condition [and it does]. Therefore, it's necessary to create a fresh copy to prevent this issue.